### ISSUE TITLE
The reproduction results of the DLPFC dataset are inconsistent with those recorded in the paper

### ISSUE THREAD

**Issue by JingofGao** [JinmiaoChenLab/GraphST#13]
My code is below.
`samples = ["151507", "151508", "151509",
           "151510", "151669", "151670",
           "151671", "151672", "151673",
           "151674", "151675", "151676"]
n_clusters = [7] * 4 + [5] * 4 + [7] * 4

ARIs = []
refinement = False
for i in range(12):
    dataset = samples[i]
    file_fold = "/media/s5/gj/nodecode/data/DLPFC/{}".format(dataset)
    adata = sc.read_visium(file_fold, count_file='{}_filtered_feature_bc_matrix.h5'.format(dataset), load_images=True)
    adata.var_names_make_unique()

    # define model
    model = GraphST.GraphST(adata, device=device)

    # train model
    adata = model.train()

    # clustering
    # For DLPFC dataset, we use optional refinement step.
    clustering(adata, n_clusters[i], radius=50, method='mclust', refinement=refinement)

    # add ground_truth
    df_meta = pd.read_csv(file_fold + '/gt/{}_truth.txt'.format(dataset), sep='\t', header=None, index_col=0)
    adata.obs['ground_truth'] = df_meta.values

    # filter out NA nodes
    adata = adata[~pd.isnull(adata.obs['ground_truth'])]

    # calculate metric ARI
    ARI = metrics.adjusted_rand_score(adata.obs['domain'], adata.obs['ground_truth'])
    ARIs.append(round(ARI, 2))

print(ARIs)
print("mean:", round(np.mean(ARIs), 2))

"""
* * * refinement = True * * *
[0.43, 0.5, 0.53, 0.47, 0.59, 0.43, 0.61, 0.64, 0.64, 0.62, 0.6, 0.43]
mean: 0.54

* * * refinement = False * * *
[0.43, 0.46, 0.48, 0.44, 0.53, 0.43, 0.58, 0.59, 0.6, 0.59, 0.54, 0.41]
mean: 0.51
"""
`

**Comment by StackChan** [comment]
You shall add preprocessing steps like
```
sc.pp.highly_variable_genes(adata, flavor="seurat_v3", n_top_genes=3000)
sc.pp.normalize_total(adata, target_sum=1e4)
sc.pp.log1p(adata)
sc.pp.scale(adata, zero_center=False, max_value=10)
```


Full code as below:
```
# use py3.9 environment
import os
import torch
import pandas as pd
import scanpy as sc
from sklearn import metrics
import multiprocessing as mp
from GraphST import GraphST
from GraphST.utils import clustering
import numpy as np

# 设置设备
device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')

# 设置R环境变量
# the location of R, which is necessary for mclust algorithm. Please replace the path below with local R installation path
# os.environ['R_HOME'] = r'D:\Software\R-4.3.1'
# # the location of R (used for the mclust clustering)
# os.environ['R_USER'] = r'D:\Software\Anaconda\envs\py3.9\Lib\site-packages\rpy2'

# 定义样本和聚类簇数量
samples = ["151507"]

n_clusters = [7] * 4 + [5] * 4 + [7] * 4
ARIs = []
refinement = True

# for i in range(12):
# for i in range(1):
dataset = samples[i]
file_fold = "/data/pub/SRT/DLPFC/{}".format(dataset)

# 读取数据集
adata = sc.read_visium(file_fold, count_file='{}_filtered_feature_bc_matrix.h5'.format(dataset), load_images=True)
adata.var_names_make_unique()

# 预处理步骤
sc.pp.highly_variable_genes(adata, flavor="seurat_v3", n_top_genes=3000)
sc.pp.normalize_total(adata, target_sum=1e4)
sc.pp.log1p(adata)
sc.pp.scale(adata, zero_center=False, max_value=10)

# define model
model = GraphST.GraphST(adata, device=device)

# train model
adata = model.train()

# clustering
# For DLPFC dataset, we use optional refinement step.
clustering(adata, n_clusters[i], radius=50, method='mclust', refinement=refinement)

# add ground_truth
df_meta = pd.read_csv('/data/pub/SRT/DLPFC/DLPFC_annotations/{}_truth.txt'.format(dataset), sep='\t', header=None, index_col=0)
adata.obs['ground_truth'] = df_meta.iloc[:, 0]

# filter out NA nodes
adata = adata[~pd.isnull(adata.obs['ground_truth'])]

# calculate metric ARI
ARI = metrics.adjusted_rand_score(adata.obs['domain'], adata.obs['ground_truth'])
ARIs.append(round(ARI, 2))

print(ARIs)
print("mean:", round(np.mean(ARIs), 2))
```

Then ARI for "151507" will boom from 0.43 to 0.53.

Some code used to compare SOTA performance can be found here:
https://github.com/JinmiaoChenLab/SEDR_analyses/blob/master/DLPFC_stLearn.py
https://github.com/yiwen-yang/MCGAE/blob/7ea1c4daedf3eee14caace294486ee1c43ecd34e/Analysis/DLPFC/GraphST.py

And you can found from these works that they do use 【different preprocessing steps】. I know it's annoying.
